pennylane/ops/op_math/prod.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the implementation of the Prod class which contains logic for
+computing the product between operations.
+"""
+from copy import copy
+from functools import reduce
+
+import numpy as np
+
+import pennylane as qml
+from pennylane import math
+from pennylane.operation import Operator, expand_matrix
+
+
+def prod(*ops, do_queue=True, id=None):
+    """Represent the tensor product (or matrix product) between operators."""
+    return Prod(*ops, do_queue=do_queue, id=id)
+
+
+def _prod(
+    mats_gen,
+):  # TODO: current multiplication always expands mats, this is not always necessary
+    """Multiply matrices together"""
+    res = reduce(
+        math.dot, mats_gen
+    )  # Inefficient method (should group by wires like in tensor class)
+    return res
```

Comment:
No need to assign to a variable.
```suggestion
    # Inefficient method (should group by wires like in tensor class)
    return reduce(math.dot, mats_gen)
```